### PR TITLE
PortForwardServiceTest : Remove use of Thread.sleep #2752

### LINKED ISSUE #2752: PortForwardServiceTest : Remove use of `Thread.sleep`
### Component

JKube Kit

### Description

We have this code in PortForwardServiceTest which seems to be forwarding a port inside Kubernetes using [Fabric8 Kubernetes Client](https://github.com/fabric8io/kubernetes-client)

https://github.com/eclipse/jkube/blob/3afc7bf1f8dbe4452826c5ec7ec58de5c9408d53/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/PortForwardServiceTest.java#L98-L100

However, I'm not sure why we have `Thread.sleep` inside the try block after initializing port forward. We don't seem to be testing anything after this wait.

It's better to just remove this unnecessary waiting, asserting whether value returned from `forwardPortAsync` is not null should suffice.

### Expected Behavior
`Thread.sleep` is removed from PortForwardServiceTest.simpleScenario

### Acceptance Criteria
- [ ] Thread.sleep` is removed from `PortForwardServiceTest.simpleScenario`
- [ ] `PortForwardServiceTest.simpleScenario` should still pass after making these changes

### Before you start :red_circle:

:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down::point_down::point_down::point_down::point_down::point_down::point_down:
Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first. Pay special attention to the ECA agreement section and the requirement to sign-off your commit.

#### Kubernetes

If you don't have a real Kubernetes cluster available (most probably), you can use Minikube or Kind to test with a local cluster.

- Minikube [get started guide](https://minikube.sigs.k8s.io/docs/start/)
- Kind (Requires Docker) [quick start](https://kind.sigs.k8s.io/docs/user/quick-start/)

#### OpenShift

If you don't have a real OpenShift cluster available (most probably), you can use [Red Hat's developer Sandbox for Red Hat OpenShift](https://developers.redhat.com/developer-sandbox). The only requirement is to have a Red Hat account.

Once you have your Sandbox environment, you'll need to download the oc tool from the cluster console.
(Press the `?` icon and from the context menu select `Command line tools`, you'll be redirected to https://$subdomain.openshiftapps.com/command-lines-tools where you'll be able to download the CLI for your platform)

### PR DESCRIPTION
**Pull request by rschulm**
## Description

Fixes #2752

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [x] I tested my code in Kubernetes
 - [x] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/jkubeio/ci)

Started new GH workflow run for https://github.com/eclipse/jkube/pull/2849 (_2024-03-25T02:36:07Z_)

:gear: [JKube E2E Tests (8414301660)](https://github.com/jkubeio/ci/actions/runs/8414301660) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#2849)](https://github.com/jkubeio/ci/actions/runs/8414301660/job/23037771157)

**Comment by rschulm** [comment]
Can someone help explain why the check failed?

**Comment by manusa** [comment]
It's a normal failure (flake).
However, you are fixing multiple issues here. You should focus only on the one that was assigned to you, or at least the one that your mentioning in the Pull Request.

Please note that `[first-timers-only](https://github.com/eclipse/jkube/labels/first-timers-only)` issues are intended for those who never contributed to this repo before. Please, don't request to fix those since you end up confusing us.

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/archive/ArchiveDecompressor.java [hunk 1]
```diff
@@ -88,8 +88,7 @@ private static void extractArchiveContents(InputStream is, File targetDirectory)
       FileUtil.cleanDirectory(targetDirectory);
     }
     FileUtil.createDirectory(targetDirectory);
-    try (ArchiveInputStream ais = new ArchiveStreamFactory().createArchiveInputStream(is)) {
-      ArchiveEntry entry;
+    try (ArchiveInputStream<ArchiveEntry> ais = new ArchiveStreamFactory().createArchiveInputStream(is)) {      ArchiveEntry entry;
```
Comment (rohanKanojia) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/archive/ArchiveDecompressor.java:91]: How is this change related to #2752 ?

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/KubernetesHelper.java [hunk 2]
```diff
@@ -440,7 +440,7 @@ public static FilterWatchListDeletable<Pod, PodList, PodResource> withSelector(N
                     log.warn("Ignoring empty values in selector expression %s", expression);
                     continue;
                 }
-                String[] valuesArray = values.toArray(new String[values.size()]);
+                String[] valuesArray = values.toArray(new String[0]);
```
Comment (rohanKanojia) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/KubernetesHelper.java:443]: how is this change related to #2752

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricher.java [hunk 3]
```diff
@@ -316,7 +316,7 @@ private List<ServicePort> extractPorts(List<ImageConfiguration> images) {
             }
 
             // Extract first port and remove first element
-            if(portsFromImageLabels == null || portsFromImageLabels.isEmpty()) {
+            if(portsFromImageLabels.isEmpty()) {
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricher.java:319]: how is this change related to #2752

File: jkube-kit/profile/src/test/java/org/eclipse/jkube/kit/profile/ProfileUtilTest.java [hunk 4]
```diff
@@ -57,7 +57,7 @@ public File getProfileDir() throws URISyntaxException {
     }
 
     @Test
-    void findProfile_whenValidProfileArg_returnsValidProfile() throws URISyntaxException, IOException {
+    void findProfile_whenValidProfileArg_returnsValidProfile() throws URISyntaxException {
```
Comment (rohanKanojia) [inline jkube-kit/profile/src/test/java/org/eclipse/jkube/kit/profile/ProfileUtilTest.java:60]: how is this change related to #2752

**Comment by rohanKanojia** [comment]
@rschulm : Please create separate pull requests for separate issues.

**Comment by rohanKanojia** [comment]
@rschulm : Sorry I didn't see https://github.com/eclipse/jkube/pull/2849#issuecomment-2024855086 before adding my comment. Could you please address Marc's comment?

**Comment by rschulm** [comment]
> It's a normal failure (flake). However, you are fixing multiple issues here. You should focus only on the one that was assigned to you, or at least the one that your mentioning in the Pull Request.
> 
> Please note that `[first-timers-only](https://github.com/eclipse/jkube/labels/first-timers-only)` issues are intended for those who never contributed to this repo before. Please, don't request to fix those since you end up confusing us.

Sorry, I meant to submit each of those changes as separate pull requests, but I guess I messed it up and they're all under the same. How can I seperate them?

**Comment by rohanKanojia** [comment]
@rschulm You can do a hard reset of your branch to `upstream/master` and then cherry-pick 0af0997a31ea7c7f282f8f4665e416db85cc2ad2 `git cherry-pick 0af0997a31ea7c7f282f8f4665e416db85cc2ad2` , this would only include these changes.

**Comment by rschulm** [comment]
Ok, I opened a new pull request for issue #2752
